lib/rspec/expectations/failure_aggregator.rb

```diff
@@ -0,0 +1,159 @@
+module RSpec
+  module Expectations
+    # @private
+    class FailureAggregator
+      attr_reader :block_label, :metadata
+
+      def aggregate
+        RSpec::Support.with_failure_notifier(self) do
+          begin
+            yield
+          rescue ExpectationNotMetError => e
+            failures << e
+          rescue Exception => e
```

Comment:
> Should comment why explicit Exception is the right class here, since it feels pretty "deep" in the stack.

Good idea; will do.

> If I have N long running aggregated failures, am I still able to ctrl+c effectively?

Assuming you're using rspec-core or some other test framework that handles that signal gracefully, yes.  rspec-core traps that interrupt and gracefully shuts down.  Since it traps the interrupt it doesn't raise an exception anymore.  I believe cucumber, minitest, etc handle ctrl-c in similar fashion.

> Or if not, how would I ever get more than one `other_error`?

You can't ever get more than one in `other_errors`.  I thought about exposing it as a scalar instead of an array, but I thought that it was better to have a uniform interface and provide both types of errors as arrays.  Plus who knows...maybe someone day we'll figure out a way to support multiple.  It's nice to make the interface an array so that we don't have to change the API if/when we later support that.

> hrmm, I actually don't understand yet how even the normal ExpectationNotMetError "resumes" after the first failure. 

It doesn't resume.  Expectation failures use `RSpec::Support.notify_failure`, and get added to the array via `call`, defined below.  Problem is, that uses a thread-local variable, which means that if you spawn a thread and use an expectation in it, it'll raise instead of invoking `call`, below.  So this is here just to handle expectation failures in threads.    The `ExpectationNotMetError` rescue is necessary for these specs to pass:

https://github.com/rspec/rspec-expectations/blob/9ad61a35b059893ff84b158fe56897e2cbb34be1/spec/rspec/expectations/failure_aggregator_spec.rb#L149-L171

Your questions here are good, and made me realize a potential improvement we could make.  Users will probably assume that the entire `aggregate_failures` block ran to completion but if an expectation failure happened in a thread or some other error was raised, it will have aborted early.  Given that...I'm thinking we should perhaps surface that to the user somehow.  Although, I want to get this wrapped up, so maybe we'll save that for a future enhancement unless you think it's essential to the feature.